<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>756</width>
    <height>672</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>SPI Master for Stepper Drives</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>30</y>
      <width>151</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Enabled Drives</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_10">
      <item>
       <widget class="QCheckBox" name="cbEnableDrive1">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Drive 1</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_11">
      <item>
       <widget class="QCheckBox" name="cbEnableDrive2">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Drive 2</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>30</y>
      <width>191</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>Drive availability</string>
    </property>
    <widget class="QFrame" name="frame_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>151</width>
       <height>101</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>72</width>
        <height>23</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QCheckBox" name="cbA4988">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>A 4988</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>94</width>
        <height>23</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QCheckBox" name="cbDRV8825">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>DRV 8825</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>94</width>
        <height>23</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QCheckBox" name="cbRAPS128">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>RAPS 128</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>153</width>
       <height>23</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_12">
      <item>
       <widget class="QCheckBox" name="cbControllerSPI0Available">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>Controller on SPI 0</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>153</width>
       <height>23</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_13">
      <item>
       <widget class="QCheckBox" name="cbControllerSPI1Available">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>Controller on SPI 1</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>250</y>
      <width>181</width>
      <height>331</height>
     </rect>
    </property>
    <property name="title">
     <string>Select microsteps </string>
    </property>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>290</y>
       <width>161</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>(applies to both drives)</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>103</width>
       <height>260</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QRadioButton" name="rbFullStep">
        <property name="text">
         <string>Full step</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rbHalfStep">
        <property name="text">
         <string>1/2 step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rbQuarterStep">
        <property name="text">
         <string>1/4 step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rbEigthStep">
        <property name="text">
         <string>1/8 step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rbSixteenthStep">
        <property name="text">
         <string>1/16 step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rb32thStep">
        <property name="text">
         <string>1/32 step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rb64thStep">
        <property name="text">
         <string>1/64 step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rb128thStep">
        <property name="text">
         <string>1/128 step</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>250</y>
      <width>321</width>
      <height>251</height>
     </rect>
    </property>
    <property name="title">
     <string>Kinematics</string>
    </property>
    <widget class="QFrame" name="frame">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>91</width>
       <height>81</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>76</width>
        <height>62</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QRadioButton" name="rbKinematics1">
         <property name="text">
          <string>Drive 1</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rbKinematics2">
         <property name="text">
          <string>Drive 2</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QPushButton" name="pbSendKinematics">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>150</y>
       <width>211</width>
       <height>81</height>
      </rect>
     </property>
     <property name="text">
      <string>Send to Drive</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>308</width>
       <height>112</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Acceleration [msteps/s²]</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="sbAcc">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="maximum">
             <number>10000</number>
            </property>
            <property name="singleStep">
             <number>100</number>
            </property>
            <property name="value">
             <number>500</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Maximum speed [msteps/s]</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="sbVel">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="maximum">
             <number>10000</number>
            </property>
            <property name="singleStep">
             <number>100</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Microsteps</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="sbSteps">
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <property name="minimum">
           <number>-1000000000</number>
          </property>
          <property name="maximum">
           <number>1000000000</number>
          </property>
          <property name="value">
           <number>500</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>250</y>
      <width>201</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string>Motor control</string>
    </property>
    <widget class="QPushButton" name="pbStart">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>80</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Start</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbStop">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>80</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop</string>
     </property>
    </widget>
    <widget class="QFrame" name="frame_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>40</y>
       <width>91</width>
       <height>81</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>76</width>
        <height>62</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QRadioButton" name="rbCtrlDrive1">
         <property name="text">
          <string>Drive 1</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rbCtrlDrive2">
         <property name="text">
          <string>Drive 2</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>100</y>
      <width>151</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Select SPI channel</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>131</width>
       <height>62</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <widget class="QRadioButton" name="rbSPI0">
        <property name="text">
         <string>SPI channel 0</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rbSPI1">
        <property name="text">
         <string>SPI channel 1</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>230</y>
      <width>711</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>580</y>
      <width>721</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>40</y>
      <width>20</width>
      <height>181</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="pbExit">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>380</y>
      <width>201</width>
      <height>171</height>
     </rect>
    </property>
    <property name="text">
     <string>Exit</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>98</width>
      <height>23</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_8">
     <item>
      <widget class="QCheckBox" name="cbspi0Open">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>SPI 0 open</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>70</y>
      <width>98</width>
      <height>23</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <widget class="QCheckBox" name="cbspi1Open">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>SPI 1 open</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>600</y>
      <width>338</width>
      <height>23</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Drive activity:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="cbDr1Active">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Drive 1 active</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="cbDr2Active">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Drive 2 active</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
